sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -235,14 +235,18 @@ class SessionCatalog(
    * Note: If the underlying implementation does not support altering a certain field,
    * this becomes a no-op.
    */
-  def alterTable(tableDefinition: CatalogTable): Unit = {
+  def alterTable(tableDefinition: CatalogTable, fromAnalyze: Boolean = false): Unit = {
```

Comment:
The additional flag parameter `fromAnalyze` looks weird for me. Why would we need to have two APIs for alter table and use a flag like this to choose among them?